packages/wrangler/src/__tests__/helpers/mock-cfetch.ts

```diff
@@ -206,6 +209,48 @@ export function setMockFetchKVGetValue(
 	kvGetMocks.set(`${accountId}/${namespaceId}/${key}`, value);
 }
 
+/**
+ * @mocked typeof fetchR2Objects
+ */
+export async function mockFetchR2Objects(
+	resource: string,
+	bodyInit: {
+		body: BodyInit | Readable;
+		headers: HeadersInit | undefined;
+		method: "PUT" | "GET";
+	}
+): Promise<Response> {
+	if (bodyInit.body instanceof Readable) {
+		bodyInit.body.destroy();
+		bodyInit.body.removeAllListeners();
+	}
+
+	if (r2GetMocks.has(resource)) {
+		const value = r2GetMocks.get(resource);
+
+		return new Response(value);
+	}
+	throw new Error(`no expected mock found for \`r2 object get\` - ${resource}`);
+}
+
+/**
+ * Mock setter for usage within test blocks, companion helper to `mockFetchR2Objects`
+ */
+export function setMockFetchR2Objects(
+	{
+		accountId = "r2-object-test-account",
+		bucketName = "r2-test-bucket",
+		objectName = "r2-test-object",
+	},
+	value?: string
+) {
+	r2GetMocks.set(
+		`/accounts/${accountId}/r2/buckets/${bucketName}/objects/${objectName}`,
+		value
+	);
+}
+
 export function unsetMockFetchKVGetValues() {
 	kvGetMocks.clear();
+	r2GetMocks.clear();
```

Comment:
This either needs to go in its own function, or this function needs renaming. Currently it only refers to KV values.

Reply:
Im thinking renaming it to `unsetSpecialMockFns`